subsys/bluetooth/mesh/cfg_cli.c

```diff
@@ -555,20 +555,14 @@ struct mod_member_list_param {
 	size_t *member_cnt;
 };
 
-static int mod_member_list_handle(struct bt_mesh_msg_ctx *ctx,
-				   struct net_buf_simple *buf, uint16_t op,
-				   bool vnd)
+static int mod_sub_list_handle(struct bt_mesh_msg_ctx *ctx, struct net_buf_simple *buf, uint16_t op,
+			       bool vnd)
 {
 	struct mod_member_list_param *param;
 	uint16_t elem_addr, mod_id, cid;
 	uint8_t status;
 	int i;
 
-	if ((vnd && buf->len < 7U) || (buf->len < 5U)) {
```

Comment:
I guess this was correct checking. What will happen if config client receives malformed reply?

Reply:
I belive that `BT_MESH_LEN_MIN(N)` in `bt_mesh_cfg_cli_op[]` will check that the messages are greater or equal to the value of N? If that is the case I belive this check is redundant as long as we assume that this function is called via model opcode handlers (which it is ATM)

Reply:
Ok, I agree that `(buf->len < 5U)` checking can be skipped. However, this part `(vnd && buf->len < 7U)` is still valid.